utils/map.go

```diff
@@ -33,8 +33,9 @@ func GetKeysToDelete(tags map[string]string, upstreamTags map[string]string) []*
 	}
 
 	var updateUntags []*string
-	for key, val := range upstreamTags {
-		if len(tags) == 0 || tags[key] != val {
+	for key := range upstreamTags {
+		_, ok := tags[key]
+		if len(tags) == 0 || !ok {
```

Comment:
If `len(tags) == 0`, then `ok` will be `false`.
```suggestion
		if !ok {
```